microsoft/testsuites/firewalld/firewalldsuite.py

```diff
@@ -0,0 +1,228 @@
+        )
+
+    total_tests = None
+    failed_tests = None
+    skipped_tests = None
+    num_regex = r"\d+ "
+
+    total_str = cat.read_with_filter(f"{log_file}", "tests were run", sudo=True)
+    m = re.search(num_regex, total_str)
+    if m:
+        total_tests = m[0]
+
+    failed_str = cat.read_with_filter(f"{log_file}", "failed unexpectedly", sudo=True)
+    m = re.search(num_regex, failed_str)
+    if m:
+        failed_tests = m[0]
+
+    skipped_str = cat.read_with_filter(f"{log_file}", "were skipped", sudo=True)
+    m = re.search(num_regex, skipped_str)
+    if m:
+        skipped_tests = m[0]
+
+    # Logs dir contains the list of failed tests.
+    current_failures: List[str] = []
+    failed_tests_list = ls.list_dir(log_dir, sudo=True)
+    for test in failed_tests_list:
+        # each entry in this list looks like
+        # "/usr/share/firewalld/testsuite/testsuite.dir/123/"
+        # using / as delimiter, we extract the test case number
+        # in this case 123
+        current_failures.append(test.split("/")[-2])
+
+    # Note: There are known failures due to missing support for `FIB`
+    # based expressions in kernel.
+    # This is a constant list of failing test cases in each run.
+    known_failures = [
+        "061",
+        "107",
+        "120",
+        "124",
+        "192",
+        "193",
+        "194",
+        "195",
+        "196",
+        "197",
+        "200",
+        "240",
+        "252",
+        "306",
+        "324",
+    ]
+
+    # Aarch64 kernel config has some additional FIB enabled
+    # configs compared to x86_64.
+    # Aarch64 failures list is thus a subset of known failures.
+    if not set(current_failures).issubset(set(known_failures)):
+        raise LisaException(
+            "Found unexpected failures, check logs for details\n"
+            f"expected failure list: {known_failures}\n"
+            f"actual failure list:   {current_failures}\n"
+        )
+
+    # Update the test result data
+    result.set_status(
+        TestStatus.PASSED,
+        f"TOTAL:{total_tests}\nFAILED:{failed_tests}\nSKIPPED:{skipped_tests}\n",
+    )
+
+
+@TestSuiteMetadata(
+    area="firewalld",
+    category="functional",
+    description="""
+    This test suite is to validate firewalld daemon on Azure Linux 3.0.
+    """,
+)
+class FirewalldSuite(TestSuite):
+    def before_case(self, log: Logger, **kwargs: Any) -> None:
+        node: Node = kwargs["node"]
+        if not isinstance(node.os, CBLMariner) or node.os.information.version < "3.0.0":
+            raise SkippedException(
+                UnsupportedDistroException(
+                    node.os, "Firewalld testsuite is supported only on AzureLinux 3.0."
+                )
+            )
+
+    @TestCaseMetadata(
+        description="""
+        This test case runs the firewalld testsuite.
+        The test suite is a collection of tests which run against
+        a local firewalld installation.
```

Comment:
Please put test suite description in TestSuiteMetadata, here put the test case description is sufficient.

Reply:
@LiliDeng Sure.